core/contracts/common/FixedPoint.sol

```diff
@@ -97,6 +127,22 @@ library FixedPoint {
         return Unsigned(a.rawValue.mul(b));
     }
 
+    /** @dev Multiplies two `Unsigned`s, reverting on overflow, and ceil's the resultant product rather than the default floor behavior. */
+    function mulCeil(Unsigned memory a, Unsigned memory b) internal pure returns (Unsigned memory) {
+        uint mulRaw = a.rawValue.mul(b.rawValue);
+        uint mod = mulRaw.mod(FP_SCALING_FACTOR);
+        uint mulFloor = mulRaw / FP_SCALING_FACTOR;
+        if (mod != 0) {
```

Comment:
An alternative to computing the mod is to:

`if (mulFloor * FP_SCALING_FACTOR != mulRaw)`

Thoughts?

Reply:
this does reduce a line of code, @mrice32 thoughts?

Reply:
We can also replace the `mod` line of code in `divCeil` with

`if (divFloor.mul(b.rawValue) != divRaw)`

Reply:
Not super opinionated either way.